sunpy/net/helio/hec.py

```diff
@@ -162,7 +166,12 @@ def search(self, *args, **kwargs):
                                                     FROM=table,
                                                     MAXRECORDS=max_records)
         results = votable_handler(etree.tostring(results))
-        return HECResponse(results.to_table(), client=self)
+        table = HECResponse(results.to_table(), client=self)
+        if len(table) == 500:
```

Comment:
```suggestion
        if len(table) == max_records == 500:
```

Reply:
So it doesn't emit if `max_records` is higher, but by chance the number of queried records is 500